server/web/server.go

```diff
@@ -76,6 +77,22 @@ func NewHttpServerWithCfg(cfg *Config) *HttpServer {
 // MiddleWare function for http.Handler
 type MiddleWare func(http.Handler) http.Handler
 
+// LifeCycleCallback configures callback.
+type LifeCycleCallback interface {
+	AfterStart(app *HttpServer)
+	BeforeShutdown(app *HttpServer)
+	mustEmbedUnimplementedLifeCycleCallback()
+}
+
+type UnimplementedLifeCycleCallback struct {
```

Comment:
I think we don't need this because if we add more function to `LifeCycleCallback` interface, it's better for users to know this. They can easily get compiler error.
And it's not possible for users to have many callbacks, so it's easy for them to change it.
This is different from gRPC because gRPC is a RPC framework so it must keep both client side and server side compatible. But Beego doen't need this.